### PR TITLE
Checker: fix exception on wrong update syntax in anon records

### PR DESCRIPTION
**Pull request by auduchinok**
Fixes exceptions introduced in https://github.com/dotnet/fsharp/pull/15214 when wrong syntax is used to update an anonymous record:

```fsharp
let f (r: {| A: int |}) =
    { r with A = 1 }
```

### CODE REVIEW

**Review by vzarytovskii — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs [hunk 1]
```diff
@@ -74,4 +74,17 @@ module NestedAnonRecds
...
+    let ``Wrong update syntax`` () =
+        Fsx """
+let f (r: {| A: int |}) =
+    { r with A = 1 }
+"""
+        |> ignoreWarnings
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 39, Line 2, Col 36, Line 2, Col 37, "The record label 'A' is not defined.")
```
Comment (T-Gro) [inline tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs]:
```suggestion
            (Error 39, Line 3, Col 14, Line 3, Col 15, "The record label 'A' is not defined.")
```

Comment (auduchinok) [reply tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs]: @T-Gro Thanks, I've pushed the fix just before it. 🙂

**Comment by KevinRansom** [comment]
nice

**Comment by vzarytovskii** [comment]
Backport is needed to resolve merge issues in 17.7 -> main, it won't go into QB.

**Pull request merged**